test/functional/tests/io/trim/test_trim.py

```diff
@@ -86,6 +87,110 @@ def test_trim_start_discard():
+
+    with TestRun.step(f"Disable udev"):
+        os_utils.Udev.disable()
+
+    with TestRun.step(f"Prepare cache instance in WB with one core"):
+        cache = casadm.start_cache(cache_dev, CacheMode.WB, force=True)
+        core = cache.add_core(core_dev)
+        cache.set_cleaning_policy(CleaningPolicy.nop)
+        cache.set_seq_cutoff_policy(SeqCutOffPolicy.never)
+        cache.purge_cache()
+
+    with TestRun.step(f"Fill exported object with dirty data"):
+        core_size_4k = core.get_statistics().config_stats.core_size.get_value(Unit.Blocks4096)
+        core_size_4k = int(core_size_4k)
+        dd = (
+            Dd()
+            .input("/dev/urandom")
+            .output(core.path)
+            .count(core_size_4k)
+            .block_size(Unit.Blocks4096)
+            .oflag("direct")
+        )
+        dd.run()
+        os_utils.sync()
+
+        dirty_4k = cache.get_statistics().usage_stats.dirty.get_value(Unit.Blocks4096)
+
+        if dirty_4k != core_size_4k:
+            TestRun.fail(
+                f"Failed to fill cache. Expected dirty blocks: {core_size_4k}, "
+                f"actual value {dirty_4k}"
+            )
+
+    with TestRun.step(f"Discard 4k of data on exported object"):
+        TestRun.executor.run_expect_success(f"blkdiscard {core.path} --length 4096 --offset 0")
+
+        old_occupancy = cache.get_statistics().usage_stats.occupancy.get_value(Unit.Blocks4096)
+
+    with TestRun.step(f"Copy contents of cache disk"):
+        exp_obj_dump_path = "/root/exp_obj_dump.tmp"
+        dd = (
+            Dd()
+            .input(core.path)
+            .output(exp_obj_dump_path)
+            .count(core_size_4k)
+            .block_size(Unit.Blocks4096)
+            .iflag("direct")
+        )
+        dd.run()
+        os_utils.sync()
+
+    with TestRun.step("Power cycle"):
+        power_control = TestRun.plugin_manager.get_plugin("power_control")
+        power_control.power_cycle()
+        os_utils.Udev.disable()
+
+    with TestRun.step("Load cache"):
+        cache = casadm.start_cache(cache_dev, load=True)
+
+    with TestRun.step("Check if occupancy after dirty shutdown is correct"):
+        new_occupancy = cache.get_statistics().usage_stats.occupancy.get_value(Unit.Blocks4096)
+        if new_occupancy != old_occupancy:
+            TestRun.LOGGER.error(
+                f"Expected occupancy after dirty shutdown: {old_occupancy}. "
+                f"Actuall: {new_occupancy})"
+            )
+
+    with TestRun.step("Compare md5"):
+        expected_md5 = TestRun.executor.run(f"md5sum {exp_obj_dump_path}").stdout.split()[0]
+        actual_md5 = TestRun.executor.run(f"md5sum {core.path}").stdout.split()[0]
+        if expected_md5 != actual_md5:
+            TestRun.LOGGER.error("md5 doesn't match")
+
+    with TestRun.step("Remove dump file"):
+        actual_md5 = TestRun.executor.run(f"rm -f {exp_obj_dump_path}").stdout
```

Comment:
This code will not be run if exception occurs in any of previous steps. If we need test-specific cleanup, a fixture is the way to go.